src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -803,20 +751,22 @@ def _validate_vmss_create_load_balancer_or_app_gateway(namespace):
 
         # AppGateway frontend
         required = []
-        if namespace.app_gateway_type == 'new' and namespace.vnet_type != 'new':
-            required.append('app_gateway_subnet_address_prefix')
+        if namespace.app_gateway_type == 'new':
+            required.append('app_gateway_sku')
+            required.append('app_gateway_capacity')
+            if namespace.vnet_type != 'new':
+                required.append('app_gateway_subnet_address_prefix')
         elif namespace.app_gateway_type == 'existing':
             required.append('backend_pool_name')
         forbidden = ['nat_pool_name', 'load_balancer']
-        _validate_network_balancer_required_forbidden_parameters(
-            namespace, required, forbidden, 'application gateway')
+        validate_parameter_set(namespace, required, forbidden, description='network balancer: application gateway')
 
     elif balancer_type == 'loadBalancer':
         # LoadBalancer frontend
         required = []
-        forbidden = ['app_gateway_subnet_address_prefix', 'application_gateway']
-        _validate_network_balancer_required_forbidden_parameters(
-            namespace, required, forbidden, 'load balancer')
+        forbidden = ['app_gateway_subnet_address_prefix', 'application_gateway', 'app_gateway_sku',
+                     'app_gateway_capacity']
+        validate_parameter_set(namespace, required, forbidden, description='network balancer: load balancer')
```

Comment:
I use network balancer because that's how it's listed in the argument group.  So the error would look like:
```
usage error for network balancer: load balancer
  missing: blahblay
  not applicable: blahblah
```